packages/playwright/src/runner/loadUtils.ts

```diff
@@ -67,14 +67,12 @@ export async function collectProjectsAndTestFiles(testRun: TestRun, doNotRunTest
   const projectClosure = buildProjectsClosure([...filesToRunByProject.keys()]);
   for (const [project, type] of projectClosure) {
     if (type === 'dependency') {
-      filesToRunByProject.delete(project);
```

Comment:
if you do: delete/set it looses the order.
if you do: set, it keeps the order in JavaScript.